### PR TITLE
feat(batch): use iceberg rust in iceberg scan

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Tracking issue: https://github.com/risingwavelabs/risingwave/issues/17548
- Use iceberg rust instead of icelake in batch crate. `IcebergScanExecutor` will only depend on iceberg-rust after this PR.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/sink/iceberg/mod.rs
Comment (xxchan) [inline src/connector/src/sink/iceberg/mod.rs:1]: Is the change on sink accidentally related? :eyes:

File: src/connector/src/sink/iceberg/mod.rs [hunk 1]
```diff
@@ -307,19 +308,32 @@ impl IcebergConfig {
             iceberg_configs.insert(CATALOG_NAME.to_string(), self.catalog_name());
 
             if let Some(region) = &self.region {
+                // icelake
                 iceberg_configs.insert(
                     "iceberg.table.io.region".to_string(),
```
Comment (fuyufjh) [inline src/connector/src/sink/iceberg/mod.rs:313]: Why need to keep the config for the deprecated `icelake` implementation?

**Review by fuyufjh — APPROVED** [review]

File: src/connector/src/sink/iceberg/mod.rs (hunk 1 above)
Comment (chenzl25) [reply src/connector/src/sink/iceberg/mod.rs:313]: Icelake is still used in other crates like `frontend` (metadata) and `connector` (e.g. sink). The frontend part would be done in a later PR, but we still need icelake for sink.

File: src/connector/src/sink/iceberg/mod.rs
Comment (chenzl25) [reply src/connector/src/sink/iceberg/mod.rs:1]: Iceberg sink can't be replaced in the short term. As far as I know, we can do it until Iceberg-rust releases `0.4.0` version. (currently `0.2.0`)

Comment (xxchan) [reply src/connector/src/sink/iceberg/mod.rs:1]: I mean, why do we change sink's code for batch?

Comment (xxchan) [reply src/connector/src/sink/iceberg/mod.rs:1]: Ohh, it seems the config is defined in sink, but shared by batch and sink... Perhaps we should refactor it later

Comment (xxchan) [reply src/connector/src/sink/iceberg/mod.rs:1]: BTW, perhaps we can create a tracking issue for deprecating icelake, especially note down blockers.

**Pull request merged**